integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/AbstractIntegrationTest.java

```diff
@@ -139,6 +141,14 @@ protected void createDefaultEnvironment(TestContext testContext) {
                 .validate();
     }
 
+    protected void createFreeipa(TestContext testContext) {
```

Comment:
Here I cannot see the point of this method. You already can here `createDefaultEnvironment` with `.withCreateFreeIpa(Boolean.FALSE)` and `initiateEnvironmentCreation` along `.withCreateFreeIpa(Boolean.TRUE)`. So I do not think this separate FreeIPA creation is needed at `AbstractIntegrationTest` class.

Reply:
That way I can not reference the created freeipa with `testContext.given(FreeIpaTestDto.class)` and I need it to be able to call the salt password rotation

Reply:
That way I can not reference the created freeipa with `testContext.given(FreeIpaTestDto.class)` and I need it to be able to call the salt password rotation on FreeIpa

Reply:
So you can create FreeIPA in the test steps here, for example: [DistroXEncryptedVolumeTest](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java), [MonitoringTests](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/l0promotion/MonitoringTests.java) or in case of [several FreeIPA E2E test cases](https://github.com/hortonworks/cloudbreak/tree/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/freeipa). Actually this is not need to the `AbstractIntegrationTest` from here.